hummingbot/connector/exchange/chainflip_lp/chainflip_lp_constants.py

```diff
@@ -0,0 +1,75 @@
+from hummingbot.core.data_type.in_flight_order import OrderState
+from hummingbot.core.api_throttler.data_types import RateLimit
+
+
+DEFAULT_DOMAIN = "chainflip_lp"
+EXCHANGE_NAME = "Chainflip"
+
+REST_RPC_URLS = {
+    "chainflip_lp":"https://mainnet-rpc.chainflip.io",
+    "chainflip_lp_testnet":"https://archive.perseverance.chainflip.io"
+    }
+WS_RPC_URLS = {
+    "chainflip_lp":"wss://mainnet-rpc.chainflip.io",
+    "chainflip_lp_testnet":"wss://archive.perseverance.chainflip.io"
+}
+WS_HEARTBEAT_TIME_INTERVAL = 30
+
+LISTENER_TIME_INTERVAL = 10
+
+MAX_REQUEST = 1000
+SECOND = 1
+MAX_ID_LEN = (2 ** 64) - 1
+
+
+# Public chainflip lp rpc methods
+ACTIVE_POOLS_METHOD = "cf_pool_environment"
+ASSET_BALANCE_METHOD = "lp_asset_balance"
+PING_METHOD = ""
+OPEN_ORDERS_METHOD = "cf_pool_order"
+PLACE_LIMIT_ORDER_METHOD = "lp_set_limit_order"
+CANCEL_LIMIT_ORDER = "lp_set_limit_order" # set
+MY_TRADES_METHOD = ""
+POOL_ORDERBOOK_METHOD = "cf_pool_orderbook"
+SUPPORTED_ASSETS_METHOD = "cf_supported_assets"
+FILLED_ORDER_METHOD = "lp_subscribe_order_fills"
+MARKET_PRICE_METHOD = "cf_pool_price"
+MARKET_PRICE_V2_METHOD = "cf_pool_price_v2"
+SCHEDULED_SWAPS = "cf_scheduled_swaps"
+
+# chainflip params
+SIDE_BUY = "buy"
+SIDE_SELL = "sell"
+
+#rate limit id
+GENERAL_LIMIT_ID = "General"
+
+
+ORDER_STATE = {
+    "PENDING": OrderState.PENDING_CREATE,
+    "NEW": OrderState.OPEN,
+    "PARTIALLY_FILLED": OrderState.PARTIALLY_FILLED,
+    "FILLED": OrderState.FILLED,
+    "PENDING_CANCEL": OrderState.PENDING_CANCEL,
+    "CANCELED": OrderState.CANCELED,
+    "REJECTED": OrderState.FAILED,
+}
+
+
+RATE_LIMITS = [
+    RateLimit(GENERAL_LIMIT_ID,MAX_REQUEST, SECOND)
+]
+
+ASSET_PRECISIONS = {
+    "Ethereum": 10e18,
+    "Bitcoin": 10e8,
+    "Polkadot": 10e12,
+    "Solana": 10e9,
+    "Litecoin": 10e9,
+    "Stable":10e6
+}
+
+STABLE_ASSETS = ["USDC","USDT"]
+FRACTIONAL_BITS = 96
```

Comment:
this is only true for `sqrt price` value, other value might use a different notation, so might make sense to call it `SQRT_PRICE_FRACTIONAL_BITS`

Reply:
Please could you tell me the different notations that exist in the rpc methods especially the one for normal price because i used this for normal price conversion as well.

Reply:
wha rpc methods did you use? the normal price (non sqrt price) should have 128 fractional bits: https://docs.chainflip.io/lp/integrations/lp-api#price